djangocms_picture/models.py

```diff
@@ -1,83 +1,284 @@
+            'Example: "720" and not "720px".'),
+    )
+    height = models.PositiveIntegerField(
+        verbose_name=_('Height'),
+        blank=True,
+        null=True,
+        help_text=_('The image height as number in pixels. '
+            'Example: "720" and not "720px".'),
+    )
+    alignment = models.CharField(
+        verbose_name=_('Alignment'),
+        choices=PICTURE_ALIGNMENT,
+        blank=True,
+        max_length=255,
+        help_text=_('Aligns the image to the selected option.'),
+    )
+    caption_text = models.TextField(
+        verbose_name=_('Caption text'),
+        blank=True,
+        help_text=_('Provide a description, attribution, copyright or other information.')
+    )
+    attributes = AttributesField(
+        verbose_name=_('Attributes'),
+        blank=True,
+        excluded_keys=['src', 'width', 'height'],
+    )
+    # link models
+    link_url = models.URLField(
+        verbose_name=_('External URL'),
+        blank=True,
+        max_length=255,
+        help_text=_('Wraps the image in a link to an external URL.'),
+    )
+    link_page = PageField(
+        verbose_name=_('Internal URL'),
+        blank=True,
+        null=True,
+        on_delete=models.SET_NULL,
+        help_text=_('Wraps the image in a link to an internal (page) URL.'),
+    )
+    link_target = models.CharField(
+        verbose_name=_('Link target'),
+        choices=LINK_TARGET,
+        blank=True,
+        max_length=255,
+    )
+    link_attributes = AttributesField(
+        verbose_name=_('Link attributes'),
+        blank=True,
+        excluded_keys=['href', 'target'],
+    )
+    # cropping models
+    # active per default
+    use_automatic_scaling = models.BooleanField(
+        verbose_name=_('Automatic scaling'),
+        blank=True,
+        default=True,
+        help_text=_('Uses the placeholder dimenstions to automatically calculate the size.'),
+    )
+    # ignores all other cropping options
+    # throws validation error if other cropping options are selected
+    use_no_cropping = models.BooleanField(
+        verbose_name=_('Use original image.'),
+        blank=True,
+        default=False,
+        help_text=_('Outputs the raw image without cropping.'),
+    )
+    # upscale and crop work together
+    # throws validation error if other cropping options are selected
+    use_crop = models.BooleanField(
+        verbose_name=_('Crop image'),
+        blank=True,
+        default=False,
+        help_text=_('Crops the image according to the given thumbnail settings in the template.'),
+    )
+    use_upscale = models.BooleanField(
+        verbose_name=_('Upscale image'),
+        blank=True,
+        default=False,
+        help_text=_('Upscales the image to the size of the thumbnail settings in the template.')
+    )
+    # overrides all other options
+    # throws validation error if other cropping options are selected
+    use_thumbnail = models.ForeignKey(
```

Comment:
misleading, if it's a relationship field then it should be called thumbnail_options.
You can create a property on the model called use_thumbnail which does `return bool(self.thumbnail_option_id)`